torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -1553,17 +1554,17 @@ std::vector<Elem> makeListForDictValues(const c10::ivalue::UnorderedMap& dict) {
 Operation dictValues(const Node* n) {
   auto outputType = n->output()->type()->expect<ListType>();
   return [=](Stack& stack) -> int {
-    auto dict = pop(stack).toGenericDictRef();
+    const auto &order = pop(stack).toGenericDict()->iterationOrder();
     if (outputType->getElementType()->isSubtypeOf(TensorType::get())) {
```

Comment:
The switching on element type can happen at compile time instead of at runtime if you return different `Operation` lambdas for each case

Reply:
Happy to take that as a trivial followup. I want to get the tests stable as soon as possible.

Reply:
Filed #19581